### DISCUSSION TITLE
Any plans for a TanStack Query wrapper in the future?

### DISCUSSION THREAD

**Discussion by austins** [icflorescu/trpc-sveltekit#37] [Announcements]
There's Svelte Query and React Query, now named TanStack Query, which will have a Svelte adapter in the future. The benefits of having a data-fetching library include implementations of caching, deduping multiple requests into a single request, infinite pagination, and observation of mutations that allow refetch on interval, mount, reconnect, window focus, etc.

Are there plans to create a thin wrapper over TanStack Query similar to @trpc/react in the future once they have that adapter? Asking in case someone else already implemented something or has plans.

**Comment by icflorescu** [comment]
That would be interesting, I'll have a look at it. Feel free to contribute if you have an idea on how to implement this.

**Comment by laneme** [comment]
They have it official now https://tanstack.com/query/v5/docs/svelte/overview

**Comment by anatolzak** [comment]
https://github.com/vishalbalaji/trpc-svelte-query-adapter
